src/lib/transaction_snark/test/account_timing/account_timing.ml

```diff
@@ -297,4 +296,349 @@ let%test_module "account timing check" =
             unchecked_timing unchecked_min_balance
       | _ ->
           false
+
+    (* user commands with timings *)
+
+    let constraint_constants = Genesis_constants.Constraint_constants.compiled
+
+    let keypairss, keypairs =
+      (* these tests are based on relative balance/payment amounts, don't need to
+         run on multiple keypairs
+      *)
+      let length = 1 in
+      (* keypair pair (for payment sender/receiver)  *)
+      let keypairss =
+        List.init length ~f:(fun _ ->
+            (Signature_lib.Keypair.create (), Signature_lib.Keypair.create ()))
+      in
+      (* list of keypairs *)
+      let keypairs =
+        List.map keypairss ~f:(fun (kp1, kp2) -> [ kp1; kp2 ]) |> List.concat
+      in
+      (keypairss, keypairs)
+
+    (* we're testing timings, not signatures *)
+    let validate_user_command uc =
+      let (`If_this_is_used_it_should_have_a_comment_justifying_it validated_uc)
+          =
+        Signed_command.to_valid_unsafe uc
+      in
+      validated_uc
+
+    let apply_user_commands_at_slot ledger slot
+        (txns : Mina_transaction.Transaction.t list) =
+      ignore
+        ( List.map txns ~f:(fun txn ->
+              let uc =
+                match txn with
+                | Command (Signed_command uc) ->
+                    uc
+                | _ ->
+                    failwith "Expected signed user command"
+              in
+              let validated_uc = validate_user_command uc in
+              match
+                Mina_ledger.Ledger.apply_user_command ~constraint_constants
+                  ~txn_global_slot:slot ledger validated_uc
+              with
+              | Ok _txn_applied ->
+                  ()
```

Comment:
We should also test that transaction snark generation doesn't fail when Ledger.apply_user_command succeeds

Reply:
@deepthiskumar What's the call to make that check?

I see `check_base_snarks` in the txn snark profiler. Something similar to that?